### PR TITLE
Variant Effect Predictor

### PR DESCRIPTION
**Pull request by viklund**

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -283,6 +284,69 @@ process intersect_files {
...
+    script:
+    """
+    case "$infile" in
+        *vcf) FORMAT=vcf ;;
+        *bed) FORMAT=ensembl ;;
+        *)    printf "Unrecognized format for '%s'" "$infile" >&2
+              exit 1;;
+    esac
+
+    OUTFILE=\$( echo $infile | sed 's/\$/.vep/' )
```
Comment (andkaha) [inline main.nf]: `OUTFILE="$infile.vep"` or `OUTFILE="\$infile.vep"`. Not quite sure what is needed here in terms of escaping, but you're not escaping `$` in `$infile` in the `case`-statement... Pulling out `sed` is overkill though.

File: main.nf [hunk 2]
```diff
@@ -283,6 +284,69 @@ process intersect_files {
...
+    }
+
+    module 'bioinfo-tools'
+    module "$params.modules.vep"
+
+    script:
+    """
+    case "$infile" in
+        *vcf) FORMAT=vcf ;;
+        *bed) FORMAT=ensembl ;;
```
Comment (andkaha) [inline main.nf]: `FORMAT="vcf"` and `FORMAT="ensembl"`, just because why not.

File: main.nf [hunk 3]
```diff
@@ -283,6 +284,69 @@ process intersect_files {
...
+        *)    printf "Unrecognized format for '%s'" "$infile" >&2
+              exit 1;;
+    esac
+
+    OUTFILE=\$( echo $infile | sed 's/\$/.vep/' )
+    ## TODO: Figure out if cache is needed
+    VEP_CACHE=/sw/data/uppnex/vep/84
```
Comment (andkaha) [inline main.nf]: `VEP_CACHE="/sw/data/uppnex/vep/84"`

File: main.nf [hunk 4]
```diff
@@ -283,6 +284,69 @@ process intersect_files {
...
+    variant_effect_predictor.pl \
+        -i $infile               \
```
Comment (andkaha) [inline main.nf]: `-i "$infile"`, or possibly `-i "\$infile"` depending on what's evaluating it (Groovy/shell).

File: main.nf [hunk 5]
```diff
@@ -283,6 +284,69 @@ process intersect_files {
...
+        --format \$FORMAT        \
+        -cache --dir \$VEP_CACHE \
+        -o \$OUTFILE             \
+        --vcf                    \
```
Comment (andkaha) [inline main.nf]: Double quote `$FORMAT`, `$VEP_CACHE` and `$OUTFILE`.

File: nextflow.config [hunk 6]
```diff
@@ -10,6 +10,7 @@ params {
         vcftools = "vcftools/0.1.14"
         tabix = "tabix/0.2.6"
         bedtools = "BEDTools/2.23.0"
+        vep = 'vep/84'
```
Comment (andkaha) [inline nextflow.config]: Double quotes here for consistency.

File: nextflow.config [hunk 7]
```diff
@@ -18,6 +19,10 @@ params {
     readlen = 150
     ref_fasta = "/sw/data/uppnex/ToolBox/ReferenceAssemblies/hg38make/bundle/2.8/b37/human_g1k_v37.fasta"
 
+    vep {
+        assembly = 'GRCh37'
```
Comment (andkaha) [inline nextflow.config]: Double quotes here for consistency.

File: main.nf [hunk 8]
```diff
@@ -283,6 +284,69 @@ process intersect_files {
...
+    publishDir params.outdir, mode: 'copy'
+
+    // We only need one core for this part
+    if ( nextflow_running_as_slurmjob() ) {
+        executor 'local'
+    }
+    else {
+        executor 'slurm'
+        queue 'core'
+        time params.short_job
```
Comment (andkaha) [inline main.nf:306]: `time "$params.short_job"` ? Not sure what's meant. Is this a Groovy variable called `params.short_job` or is it a variable called `params`?  A `$` seems to be missing. I'm not Groovy enough...

File: main.nf [hunk 9]
```diff
@@ -283,6 +284,69 @@ process intersect_files {
...
+    }
+
+    module 'bioinfo-tools'
+    module "$params.modules.vep"
+
+    script:
+    """
```
Comment (kusalananda) [inline main.nf]: Maybe use `infile="$infile"` and then use `"\$infile"` throughout below?

File: main.nf [hunk 10]
```diff
@@ -283,6 +284,69 @@ process intersect_files {
...
+        --sift b                 \
+        --polyphen b             \
+        --symbol                 \
+        --numbers                \
+        --biotype                \
+        --total_length           \
+        --canonical              \
+        --ccds                   \
+        --fields Consequence,Codons,Amino_acids,Gene,SYMBOL,Feature,EXON,PolyPhen,SIFT,Protein_position,BIOTYPE \
+        --assembly $params.vep.assembly \
```
Comment (andkaha) [inline main.nf]: `--assembly "$params.vep.assembly"`

File: main.nf [hunk 11]
```diff
@@ -283,6 +284,71 @@ process intersect_files {
...
+        queue 'core'
+        time params.short_job
+    }
+
+    module 'bioinfo-tools'
+    module "$params.modules.vep"
+
+    script:
+    """
+    infile=$infile
```
Comment (andkaha) [inline main.nf]: `infile="$infile"`

**Pull request merged**